tts-script-generation.md

```diff
@@ -0,0 +1,77 @@
+- **Normalization**: Normalization is the process of converting numbers, abbreviations, datetime, units, etc into their spoken form. Normalization is required to reduce ambiguity between the characters in the script eventually reducing the ambiguity in the input data to the model. For English, there is a normalization script in Nemo, but it won’t catch everything, and a manual review is required.
+- **Grammar**: Each line of text that is a question should end with a question mark. This is so the model can learn the pronunciation of sentences that are questions (they often rise in pitch at the end). Other end-of-line punctuation - periods, exclamation marks - is less important, but valuable.
+
+
+## Heuristics for manually normalizing
+- Search for each digit 0, 1, 2, 3, 4, 5, 6, 7, 8, 9. They will be present if there are normalization errors. Manually normalize them into written form - for example, if you see “1950s”, normalize them to “nineteen fifties”.
+    - Sports scores often require significant manual normalization. The normalization required can vary by sport, with some sports having specialized terms like “deuce” for tennis. For example, there are a lot of sports scores in the dataset, and you need to know which sport they are to do the correct normalization of cardinals into spoken form.
+- Search for the phrase “dot”. This occurs frequently in normalization errors. Correct as appropriate. Sometimes time tokens like “5.00a.m.” will be normalized as “five dot a dot m”. This is not how the time is pronounced, and it needs to be fixed, for example, to “five a m”.
+- Search for the phrase “comma”. This can be introduced through (incorrect) programmatic normalization, similar to “dot”.  
+- Search for currency symbols such as “$”. If normalization errors have occurred, the symbols might not have been normalized into dollars, cents, pounds etc.
+- Search for the degree symbol “°”. This is often not caught in normalization and has to be manually normalized.
+- Search for the character “/” (slash). It is often not normalized properly, and you want words like “bar/lounge” pronounced as “bar lounge”, so remove the slash.
+- Capitalized proper nouns like New York City instead of new york city, or Paris instead of Paris, make it a lot easier for the voice actor to read the script.
+- Ensure questions end with a question mark. A good heuristic for this is searching for pipe, and then a question word (as the pipe | will precede the start of the sentence) - such as how, what, when, where, do you etc.
+- Search for calendar months - January”, Februaryy” etc - and ensure that the names of the months are capitalized - this makes the text easier for the voice actors to read.
+- Search for three periods (ellipsis) “...” as this is often used to denote pauses in speech. It’s often ambiguous to pronounce.
+- Capitalization of acronyms - acronyms such as “GPU” and “CPU” should be left in capitals, but space should be added between them - such as “G P U”.
```

Comment:
These are initialisms not acronyms. They should be handled consistently throughout the script. The choice of `“GPU” and “CPU” should be left in capitals, but space should be added between them` is only one option and not a hard requirement.

Reply:
I will this change to initialism.

Reply:
This isn't a hard requirement. Can you phrase so that it is more of a suggestion?